libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/TemplateEngineLanguageGenerator.cs

```diff
@@ -89,7 +89,16 @@ public TemplateEngineLanguageGenerator(string filePath, Dictionary<string, IList
         /// <returns>generated text.</returns>
         public override Task<object> GenerateAsync(DialogContext dialogContext, string template, object data, CancellationToken cancellationToken = default)
         {
-            var lgOpt = new EvaluationOptions() { Locale = dialogContext.GetLocale() };
+            var lgOpt = new EvaluationOptions()
+            {
+                Locale = dialogContext.GetLocale(),
+            };
+
+            var testOptions = dialogContext.State.GetValue<Dictionary<string, object>>("Conversation.TestOptions");
+            if (testOptions != null)
+            {
+                lgOpt.Properties = testOptions;
+            }
```

Comment:
I'm not opposed to this, but I suspect that you do not need to copy this into lgOptions. Every place has access to Conversation.TestOptions, right?  If someone wanted to dynamically set this that is where they would do a setProperty.